### ISSUE TITLE
`bg_map_cell []` not aligned to 4, accesses weird indices

### ISSUE THREAD

**Issue by copyrat90** [GValiente/butano#24]
## Test image (Unfortunately, GitHub doesn't support *.bmp files)
`graphics/dynamic_nums.bmp`

![dynamic_nums](https://user-images.githubusercontent.com/34793045/188294377-5a84a506-c92c-4694-8050-f4ee757c3a66.png)

## Code
```cpp
#include "bn_core.h"
#include "bn_log.h"
#include "bn_regular_bg_item.h"
#include "bn_regular_bg_map_cell_info.h"
#include "bn_regular_bg_map_ptr.h"
#include "bn_regular_bg_ptr.h"

// graphics/dynamic_nums.bmp
#include "bn_regular_bg_items_dynamic_nums.h"

const bn::regular_bg_item& dynamic_nums = bn::regular_bg_items::dynamic_nums;

struct dynamic_nums_bg
{
    char padding[2];
    /* alignas(4) */ bn::regular_bg_map_cell cells[1024];
    bn::regular_bg_map_item map_item;
    bn::regular_bg_item bg_item;
    bn::regular_bg_ptr bg;
    bn::regular_bg_map_ptr bg_map;

    dynamic_nums_bg()
        : cells{}, map_item(cells[0], bn::size(32, 32)),
          bg_item(dynamic_nums.tiles_item(), dynamic_nums.palette_item(), map_item),
          bg(bg_item.create_bg(128 - 120, 128 - 80)), bg_map(bg.map())
    {
        BN_LOG("alignof(bn::regular_bg_map_cell) == ", alignof(bn::regular_bg_map_cell));
        BN_LOG("alignof(cells) == ", alignof(cells)); // GNU Extension warning, but it's nice to show the error.
        set_cells_in_order();
    }

    void set_cells_in_order()
    {
        for (int y = 0; y < 4; ++y)
            for (int x = 0; x < 32; ++x)
            {
                const int cell_index = map_item.cell_index(x, y);
                bn::regular_bg_map_cell& cell = cells[cell_index];
                bn::regular_bg_map_cell_info cell_info(cell);
                cell_info.set_tile_index(cell_index);
                cell = cell_info.cell();
            }

        bg_map.reload_cells_ref();
    }
};

int main()
{
    bn::core::init();

    dynamic_nums_bg bg;

    while (true)
        bn::core::update();
}
```

# Results

## Behavior WITHOUT alignas(4)

```
[WARN] GBA Debug:	alignof(bn::regular_bg_map_cell) == 2
[WARN] GBA Debug:	alignof(cells) == 2
```
![without_alignas](https://user-images.githubusercontent.com/34793045/188294643-68a06bba-4532-4d9a-9e16-f267afd9bdf1.png)

## Behavior WITH alignas(4)

```
[WARN] GBA Debug:	alignof(bn::regular_bg_map_cell) == 2
[WARN] GBA Debug:	alignof(cells) == 4
```
![with_alignas](https://user-images.githubusercontent.com/34793045/188294653-1cc3ba99-8e5a-4e5b-b58c-e6826bd285f4.png)

**Comment by copyrat90** [comment]
While I didn't test it, I think `bn::affine_bg_map_cell` also have the same issue.

**Comment by GValiente** [comment]
[bn::aligned](https://gvaliente.github.io/butano/group__memory.html#ga3e6c93fe7f9ba3d81b89c2c87ccdee74) should be fixed in [12.0.0](https://github.com/GValiente/butano/releases/tag/12.0.0), so if you try to create a map with unaligned map cells, an error message should appear.

**Comment by copyrat90** [comment]
Great, it works nicely for both the test code, and my actual project.
![align_assertion](https://user-images.githubusercontent.com/34793045/188400405-9d03ca75-e7cc-4eda-9d34-7a210d04278f.png)


Thanks a lot!
